airflow/cli/commands/task_command.py

```diff
@@ -596,6 +597,10 @@ def task_test(args, dag=None):
         task, args.map_index, exec_date_or_run_id=args.execution_date_or_run_id, create_if_necessary="db"
     )
 
+    xcoms = args.xcoms or []
+    _validate_injected_xcoms(xcoms, task)
```

Comment:
If we stick to the `ti.inject_xcoms` approach, then it may be considered moving the validation logic to `ti.inject_xcoms` itself. It'll imply that if `inject_xcoms` is called, it should inject all the required XComs at once, but probably that's fine.

Also the exception raised from there will lack the `"Please, pass them via the --xcoms argument (see --help for more details)."` part (in the context of the method we won't know where it's called from). It will need either to be caught and added in `task_command` or omitted.

Reply:
> Also the exception raised from there will lack the [context]

You can raise something other than AirflowException (I don’t see why you need to raise that exception), and catch it here to provide the additional context.

Reply:
> I don’t see why you need to raise that exception

Let me clarify if you are questioning why we should raise an exception at all (like not go with returning a bool flag or something) or you're not sure if we need the `AirflowException` class here? If it's the latter, what is the intended scenario for that? I just noticed that it had been used for the args validation in the module.

But generally absolutely aligned! Actually I also mentioned that it may've been intercepted and enriched, definitely not a blocker, just a consideration.